src/ui/components/App.js

```diff
@@ -45,23 +46,43 @@ function setTheme(theme) {
 }
 
 class App extends React.Component {
+  statusRef = React.createRef();
+
   state = {
     orientation: "bottom",
     tooltip: null,
+    recordingId: null,
+    recordingInput: null,
   };
 
   componentDidMount() {
     const { theme } = this.props;
     setTheme(theme);
+
+    const params = new URLSearchParams(document.location.search.substring(1));
+    this.setState({
+      recordingId: params.get("id"),
+    });
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     const { theme } = this.props;
+    const { recordingId, recordingInputError } = this.state;
```

Comment:
Trying to have a better error message, else will remove this unnecessary additional state variable `recordingInputError`.